.github/workflows/functional.yml

```diff
@@ -0,0 +1,136 @@
+name: Functional Testing
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+  pull_request:
+
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+jobs:
+  functional-tests:
+    name: Gateway Functional Tests
+    permissions:
+      contents: write # for docker/build-push-action to read repo content
+    runs-on: ubuntu-22.04
+    strategy:
+        matrix:
+          k8s-version: ["latest"]
```

Comment:
If we only have one element in the matrix, we don't need this entry, we can just omit the `k8s-version` conditional expression below in the `Deploy Kubernetes` step.

We may want to run the functional tests on both the oldest supported and latest versions though like we do in the conformance tests - @pleshakov @sjberman thoughts?

Reply:
Yeah, we probably should.

Reply:
will update as per discussions on this thread.